pkg/apis/extensions/v1beta1/conversion.go

```diff
@@ -65,400 +59,3 @@ func Convert_v1beta1_ScaleStatus_To_autoscaling_ScaleStatus(in *extensionsv1beta
-	if err := s.Convert(&in.PodSelector, &out.PodSelector, 0); err != nil {
-		return err
-	}
-	out.Ingress = make([]networking.NetworkPolicyIngressRule, len(in.Ingress))
-	for i := range in.Ingress {
-		if err := Convert_v1beta1_NetworkPolicyIngressRule_To_networking_NetworkPolicyIngressRule(&in.Ingress[i], &out.Ingress[i], s); err != nil {
-			return err
-		}
-	}
-	out.Egress = make([]networking.NetworkPolicyEgressRule, len(in.Egress))
-	for i := range in.Egress {
-		if err := Convert_v1beta1_NetworkPolicyEgressRule_To_networking_NetworkPolicyEgressRule(&in.Egress[i], &out.Egress[i], s); err != nil {
-			return err
-		}
-	}
-	if in.PolicyTypes != nil {
-		in, out := &in.PolicyTypes, &out.PolicyTypes
-		*out = make([]networking.PolicyType, len(*in))
-		for i := range *in {
-			if err := s.Convert(&(*in)[i], &(*out)[i], 0); err != nil {
-				return err
-			}
-		}
-	}
-	return nil
-}
-
-func Convert_networking_NetworkPolicySpec_To_v1beta1_NetworkPolicySpec(in *networking.NetworkPolicySpec, out *extensionsv1beta1.NetworkPolicySpec, s conversion.Scope) error {
-	if err := s.Convert(&in.PodSelector, &out.PodSelector, 0); err != nil {
-		return err
-	}
-	out.Ingress = make([]extensionsv1beta1.NetworkPolicyIngressRule, len(in.Ingress))
-	for i := range in.Ingress {
-		if err := Convert_networking_NetworkPolicyIngressRule_To_v1beta1_NetworkPolicyIngressRule(&in.Ingress[i], &out.Ingress[i], s); err != nil {
-			return err
-		}
-	}
-	out.Egress = make([]extensionsv1beta1.NetworkPolicyEgressRule, len(in.Egress))
-	for i := range in.Egress {
-		if err := Convert_networking_NetworkPolicyEgressRule_To_v1beta1_NetworkPolicyEgressRule(&in.Egress[i], &out.Egress[i], s); err != nil {
-			return err
-		}
-	}
-	if in.PolicyTypes != nil {
-		in, out := &in.PolicyTypes, &out.PolicyTypes
-		*out = make([]extensionsv1beta1.PolicyType, len(*in))
-		for i := range *in {
-			if err := s.Convert(&(*in)[i], &(*out)[i], 0); err != nil {
-				return err
-			}
-		}
-	}
-	return nil
-}
-
-func Convert_v1beta1_NetworkPolicyIngressRule_To_networking_NetworkPolicyIngressRule(in *extensionsv1beta1.NetworkPolicyIngressRule, out *networking.NetworkPolicyIngressRule, s conversion.Scope) error {
-	out.Ports = make([]networking.NetworkPolicyPort, len(in.Ports))
-	for i := range in.Ports {
-		if err := Convert_v1beta1_NetworkPolicyPort_To_networking_NetworkPolicyPort(&in.Ports[i], &out.Ports[i], s); err != nil {
-			return err
-		}
-	}
-	if in.From != nil {
-		out.From = make([]networking.NetworkPolicyPeer, len(in.From))
-		for i := range in.From {
-			if err := Convert_v1beta1_NetworkPolicyPeer_To_networking_NetworkPolicyPeer(&in.From[i], &out.From[i], s); err != nil {
-				return err
-			}
-		}
-	}
-	return nil
-}
-
-func Convert_networking_NetworkPolicyIngressRule_To_v1beta1_NetworkPolicyIngressRule(in *networking.NetworkPolicyIngressRule, out *extensionsv1beta1.NetworkPolicyIngressRule, s conversion.Scope) error {
-	out.Ports = make([]extensionsv1beta1.NetworkPolicyPort, len(in.Ports))
```

Comment:
This results in ports always being non-nil. The auto-generated method preserves nil if the input is nil.